docs/DeveloperGuide.md

```diff
@@ -1,34 +1,384 @@
+
+Its main role is:
+
+* Create unique exceptions thrown by different components to signal the Main function what error has occured
+
+
+## **Implementation**
+
+This section explains the implementations of Zoomaster features. It goes through the step-by-step proccess, expected outcomes of each feature and the design considerations.
+
+### Bookmark, Timetable and Planner modes feature (TYS)
+
+Zoomaster has three modes for users to interact in. First, bookmark mode has the list of bookmarks with links to online resources. 
+Secondly, timetable mode has a list of timetable slots. Lastly, planner mode which helps users plan their timetable. 
+To simplify input commands for users, all lists has the same keywords for adding, deleting, and showing items in the lists. 
+Hence by having seperating both list into different modes allows both lists to access the same keywords without causing conflicts when parsing commands.
+
+#### Implementation
+
+This feature extends Command class with a way to toggle between different modes of Zoomaster. The integer variable used to control the modes is stored in the Parser class called "programMode". Additionally, it implements the following operation:
+* getModeFromCommand() - Decodes the command sent by the users to figure out which mode the user wants to move to.
+
+Given below is a sequential diagram of how changing between modes occur.
+
+![](https://github.com/TYS0n1/tp/blob/team-Branch2/docs/diagrams/ChangeModeCommand%20seq%20dia.png?raw=true) <br/><br/>
+
+*<center/>Figure 2.1 Sequential diagram for ChangeModeCommand</center> <br/></br>*
+
+
+1. When Zoomaster gets a command from the user to change modes, a new ChangeModeCommand object is created.
+
+2. The ChangeModeCommand passes the command through getModeFromCommand() function to decode the mode the user wishes to change to.
+
+3. Zoomaster now executes the command and changes to the respective mode. If an invalid mode was given by the user or if the input field was empty, the execute function throws an exception and tells the user valid modes for Zoomaster.
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![](https://github.com/TYS0n1/tp/blob/team-Branch2/docs/diagrams/activity%20diagram%20change%20mode%20command.png?raw=true) 
+
+*<center/> Figure 2.2 Activity diagram for ChangeModeCommand </center> <br/></br>*
+
+#### Design consideration:
+
+##### Aspect: How to store programMode variable for security
+
+* **Alternative 1 (Current choice):** No security
+    * Pros: Easy to implement
+    * Cons: May introduce errors to the App if variable is changed outside of ChangeModeCommand class objects.
+* **Alternative 2:** Private variable and implement mode changing inside Parser class
+    * Pros: Ensure the App does not change modes outside commands to change modes.
+    * Cons: Reduces OOP standard of code by decoupling ChangeModeCommand from Command class and increases code complexity.
```

Comment:
Try adding some conclusion after listing the 2 alternatives on why you guys prioritize its pros and cons over the other?